apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+
+	middleware "github.com/go-swagger/go-swagger/httpkit/middleware"
+	"github.com/vmware/vic/apiservers/portlayer/models"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/operations"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/operations/interaction"
+	"github.com/vmware/vic/apiservers/portlayer/restapi/options"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/portlayer/attach"
+	"github.com/vmware/vic/portlayer/network"
+)
+
+// ExecHandlersImpl is the receiver for all of the exec handler methods
+type InteractionHandlersImpl struct {
+	attachServer *attach.Server
+}
+
+var (
+	interactionSession = &session.Session{}
```

Comment:
The interaction portion of the portlayer doesn't touch vSphere at all, so should not need a session object.